contracts/ROSCA.sol

```diff
@@ -245,7 +251,7 @@ contract ROSCA {
    * user with winnings
    */
   function cleanUpPreviousRound() internal {
-    uint256 winnerIndex;
+    uint256 winnerIndex = membersAddresses.length + 1; // to avoid default (0) to be winner
```

Comment:
also, after reading the clean code, please refactor cleanUpPreviousRound and swapWinner. Their logic is dangled. Specially `swapWinner` looks bad because the winnerIndex is passed in but only used when winnerSelectedThroughBid is false.

Why we introduce another variable `winnerSelectedThroughBid` when we have typesOfROSCA.AUCTION_ROSCA?

Reply:
i am confused with the first statement about `winnerIndex` only being used when `winnerSelectedThroughBid` is false, if you look at https://github.com/shine2lay/rosca-contracts/blob/2fcd2521fa35d06ffaa62bd1f813d4c87051c4e4/contracts/ROSCA.sol#L370. you can see it winnerIndex is being used regardless of the `winnerSelectedThroughBid`

Reply:
that being said, i think you are absolutely right in that there were alot tangled logic, i did some refactoring to clean it up